python/lsst/ip/isr/isrFunctions.py

```diff
@@ -444,3 +445,46 @@ def overscanCorrection(ampMaskedImage, overscanImage, fitType='MEDIAN', order=1,
         raise pexExcept.Exception('%s : %s an invalid overscan type' % \
             ("overscanCorrection", fitType))
     ampImage -= offImage
+
+
+def attachTransmissionCurve(exposure, opticsTransmission=None, filterTransmission=None,
+                            sensorTransmission=None, atmosphereTransmission=None):
+    """Attach a TransmissionCurve to an Exposure, given separate curves for different components.
+
+    Parameters
+    ----------
+    exposure: `lsst.afw.image.Exposure`
+        Exposure object to modify by attaching a the product of all given
+        TransmissionCurves in post-assembly trimmed detector coordinates. Must
+        have a valid Detector attached that matches to the detector associated
+        with sensorTransmission.
+    opticsTransmission: `lsst.afw.image.TransmissionCurve`
+        A TransmissionCurve that represents the throughput of the optics, to be
+        evaluated in focal-plane coordinates.
+    filterTransmission: `lsst.afw.image.TransmissionCurve`
+        A TransmissionCurve that represents the throughput of the filter
+        itself, to be evaluated in focal-plane coordinates.
+    sensorTransmission: `lsst.afw.image.TransmissionCurve`
+        A TransmissionCurve that represents the throughput of the sensor
+        itself, to be evaluated in post-assembly trimmed detector coordinates.
+    atmosphereTransmission: `lsst.afw.image.TransmissionCurve`
+        A TransmissionCurve that represents the throughput of the atmosphere,
+        assumed to be spatially constant.
+
+    Returns the TransmissionCurve attached to the exposure.
```

Comment:
Returns
    -------
    combined : `lsst.afw.image.TransmissionCurve`
        The combined transmission curve.